lib/types/message.js

```diff
@@ -1,347 +1,191 @@
-  return new Footer(describedType.value);
-};
-
-module.exports.Footer = Footer;
-
-
-
-/**
- *
- * @param data
- * @constructor
- */
-function Data(data) {
-  Data.super_.call(this, Data, data);
-}
-
-util.inherits(Data, DescribedType);
-
-Data.prototype.Descriptor = { code: 0x75, name: 'amqp:data:binary' };
-Data.fromDescribedType = function(describedType) {
-  return new Data(describedType.value);
-};
-
-module.exports.Data = Data;
-
-
-
-/**
- *
- * @param values
- * @constructor
- */
-function AMQPSequence(values) {
-  AMQPSequence.super_.call(this, AMQPSequence, values);
-}
-
-util.inherits(AMQPSequence, DescribedType);
-
-AMQPSequence.prototype.Descriptor = { code: 0x76, name: 'amqp:amqp-sequence:list' };
-AMQPSequence.fromDescribedType = function(describedType) {
-  return new AMQPSequence(describedType.value);
-};
-
-module.exports.AMQPSequence = AMQPSequence;
-
+var isData = function(x) { return x instanceof Buffer; };
+defineSection({ code: 0x75, name: 'amqp:data:binary' }, {
+  encode: null,
+  decode: function(message, described) {
+    if (!message.hasOwnProperty('body')) message.body = [];
+    else if (!Array.isArray(message.body) || !message.body.every(isData)) {
+      throw new errors.MalformedPayloadError('Attempt to put both Data and non-Data payloads in message body');
+    }
 
+    message.body.push(described.value);
+  }
+});
+
+var isSequence = function(x) { return Array.isArray(x); };
+defineSection({ code: 0x76, name: 'amqp:amqp-sequence:list' }, {
+  encode: null,
+  decode: function(message, described) {
+    if (!message.hasOwnProperty('body')) message.body = [];
+    else if (!Array.isArray(message.body) || !message.body.every(isSequence)) {
+      throw new errors.MalformedPayloadError('Attempt to put both AMQPSequence and non-AMQPSequence payloads in message body');
+    }
+
+    message.body.push(described.value);
+  }
+});
 
-/**
- *
- * @param value
- * @constructor
- */
-function AMQPValue(value) {
-  AMQPValue.super_.call(this, AMQPValue, value);
-}
+defineSection({ code: 0x77, name: 'amqp:value:*' }, {
+  encode: null,
+  decode: function(message, described) {
+    if (message.body && Array.isArray(message.body)) {
+      throw new errors.MalformedPayloadError('Attempt to provide more than one AMQPValue for message body');
+    }
 
-util.inherits(AMQPValue, DescribedType);
+    message.body = described.value;
+  }
+});
+
+defineCompositeSection({
+  name: 'header', code: 0x70,
+  fields: [
+    { name: 'durable', type: 'boolean', default: false },
+    { name: 'priority', type: 'ubyte', default: 4 },
+    { name: 'ttl', type: 'milliseconds' },
+    { name: 'firstAcquirer', type: 'boolean', default: false },
+    { name: 'deliveryCount', type: 'uint', default: 0 }
+  ]
+});
+
+defineCompositeSection({
+  name: 'properties', code: 0x73,
+  fields: [
+    { name: 'messageId', type: '*' },
+    { name: 'userId', type: 'binary' },
+    { name: 'to', type: 'string' },
```

Comment:
yeah so there are a number of these sorts of situations, where the type is `*` but thats just pointing to a `requires`, which in this case is `requires="address"`

`address` is defined as: `<type name="address-string" class="restricted" source="string" provides="address"/>`

I did provide a way to make aliases in the type system `types.js` but some of these I just got lazy on.